packages/EDA/src/tests/utils.ts

```diff
@@ -0,0 +1,121 @@
+import * as grok from 'datagrok-api/grok';
+import * as ui from 'datagrok-api/ui';
+import * as DG from 'datagrok-api/dg';
+
+const TRESHOLD = 0.5;
+const SHIFT = 1;
+const LIMIT = 2;
+
+/** Check lengths of columns */
+function checkLen(target: DG.Column, prediction: DG.Column): void {
+  if (target.length !== prediction.length)
+    throw new Error(`Non-equal elements count: ${target.length} vs. ${prediction.length}`);
+}
+
+/** Return dataframe for testing regression & linear methods */
+export function regressionDataset(samples: number, features: number, dependent: number): DG.DataFrame {
+  // create main features
+  const df = grok.data.demo.randomWalk(samples, features);
+  const cols = df.columns;
+
+  const coefs = new Float32Array(features);
+
+  // main features raw data
+  const raw = new Array<Float32Array>(features);
+  for (let j = 0; j < features; ++j)
+    raw[j] = cols.byIndex(j).getRawData() as Float32Array;
+
+  // dependent features raw data
+  for (let j = 0; j < dependent; ++j) {
+    const arr = new Float32Array(samples);
+
+    // generate coefficients
+    for (let k = 0; k < features; ++k)
+      coefs[k] = Math.random();
+
+    for (let i = 0; i < samples; ++i) {
+      for (let k = 0; k < features; ++k)
+        arr[i] += coefs[k] * raw[k][i];
+    }
+
+    cols.add(DG.Column.fromFloat32Array(`y${j}`, arr));
+  }
+
+  return df;
+} // pcaTestDf
+
+/** Max absolute deviation of the column */
+export function madNorm(col: DG.Column): number {
+  let mad = 0;
+  const rows = col.length;
+  const raw = col.getRawData();
+
+  for (let i = 0; i < rows; ++i)
+    mad = Math.max(mad, Math.abs(raw[i]));
+
+  return mad;
+}
+
+/** Max absolute deviation error */
+export function madError(target: DG.Column, prediction: DG.Column): number {
```

Comment:
understood the purpose. but why max? it doesnt accumulate info over all rows, taking mean absolute or mean squared seems more natural (and also won't confuse naming, MAD stands for mean in pandas for instance)